test/r_tree/r_tree_test.dart

```diff
@@ -226,6 +228,8 @@ main() {
         items = tree.search(Rectangle(0, 0, 50, 50));
         expect(items.length, equals(0));
 
+        expect(tree.currentRootNode, isA<LeafNode<dynamic>>());
```

Comment:
This assertion effectively replaces the `converting an empty NonLeafNode to a LeafNode` conversion test in `non_leaf_node_test.dart`.